storage: document new '/dev/disk/by-id/coreos-boot-disk' symlink (#362)
coreos/fedora-coreos-docs: modules/ROOT/pages/storage.adoc

Many of the examples below will reference a block device, such as `/dev/vda`.  The name of the available block devices depends on the underlying infrastructure (bare metal vs cloud), and often the specific instance type.  For example in AWS, some instance types have NVMe drives (`/dev/nvme*`), others use `/dev/xvda*`.

> In most cases, the simplest way to refer the target block device is using the `/dev/disk/by-id/coreos-boot-disk` link, which is created automatically and available only during machine provisioning.

Reviewer: I think maybe we should add a little more about what this link is. Maybe something like:

```suggestion
If your disk configuration is simple and uses the same disk the OS was booted from then the `/dev/disk/by-id/coreos-boot-disk` link can be used to conveniently refer to that device. This link is only available during provisioning for the purpose of making it easy to refer to the same disk the OS was booted from.
```

> Otherwise you will need to look for relevant documentation for your infrastructure around this, and/or by booting a single machine with an Ignition configuration that just gives you SSH access, and inspect the block devices via e.g. the `lsblk` command.

Reviewer: If the above suggestion is accepted we'll want to do something like:

```suggestion
In cases where you need to access other disks, the simplest thing to do is to boot a single machine with an Ignition configuration that just gives you SSH access, and inspect the block devices via e.g. the `lsblk` command.
```

version: 1.4.0
storage:
  disks:

>   - # The link to the primary block device. In virtio-based setups, this points
>     # likely to `/dev/vda`, elsewhere, likely to `/dev/sda`.

Reviewer: ```suggestion
  - # The link to the block device the OS was booted from.
```
